### PR TITLE
PLT-8402: Other users see deleted GIF previews in reply threads

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Added handling for post state `POST_DELETED` to `rhs_comment.jsx`

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-8402

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/rhs_comment.jsx [hunk 1]
```diff
@@ -441,6 +443,32 @@ export default class RhsComment extends React.Component {
...
+                    post={post}
+                    previewCollapsed={this.props.previewCollapsed}
+                    previewEnabled={this.props.previewEnabled}
+                    isEmbedVisible={this.props.isEmbedVisible}
+                >
+                    {messageWrapper}
+                </PostBodyAdditionalContent>
+            );
+        }
+
```
Comment (MusikPolice) [inline components/rhs_comment.jsx]: This code was stolen wholesale from [post_body.jsx|https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_body/post_body.jsx#L221]. Looks like we just forgot to handle this case when implementing post previews in the RHS

File: components/rhs_comment.jsx [hunk 2]
```diff
@@ -39,7 +40,8 @@ export default class RhsComment extends React.Component {
         removePost: PropTypes.func.isRequired,
         previewCollapsed: PropTypes.string.isRequired,
         previewEnabled: PropTypes.bool.isRequired,
-        isEmbedVisible: PropTypes.bool
+        isEmbedVisible: PropTypes.bool,
+        pluginPostTypes: PropTypes.object
```
Comment (MusikPolice) [inline components/rhs_comment.jsx]: This is used by the code that came from [post_body.jsx](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_body/post_body.jsx#L221). I couldn't find any cases where the prop is ever set when that component is used, so adding it as an optional prop to this component won't hurt anything. Not sure why we expose an unused prop though.

**Comment by jasonblais** [comment]
@MusikPolice Is this ready for dev review?

**Comment by MusikPolice** [comment]
@jasonblais Yup. It's just waiting on @jwilander to take a pass at it before it can be merged.

**Comment by jwilander** [comment]
I'll look shortly

File: components/rhs_comment.jsx (hunk 2 above)
Comment (jwilander) [reply components/rhs_comment.jsx]: It's just used to detect if this post type is being overridden by a plugin. Plugins don't support replies being overridden yet so you can just remove it plus the `hasPlugin` check below. I'll re-add when plugins get support for overriding replies

**Comment by MusikPolice** [comment]
@jwilander fixed.

**Pull request merged**